src/CodeMirrorBlocks.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ToggleEditor from './ui/ToggleEditor';
+import type {ToggleEditorProps} from './ui/ToggleEditor';
+import Args from './components/Args';
+import {DropTarget} from './components/DropTarget';
+import Node from './components/Node';
+import {AST} from './ast';
+import * as Nodes from './nodes';
+import * as NodeSpec from './nodeSpec';
+import * as Languages from './languages';
+import Pretty from 'pretty-fast-pretty-printer';
+import { PrimitiveGroup } from './parsers/primitives';
+
+type CodeMirrorBlocksOptions = {
+  value?: string;
+};
+
+// Consumes a DOM node to host the editor, a language object and the code
+// to render. Produces an object-representation of CMB, allowing for
+// integration with external (non-react) code
+export default class CodeMirrorBlocks {
+  constructor(container: Element, options: CodeMirrorBlocksOptions = {}, language: ToggleEditorProps['language'], cmOptions = {}) {
+    let api = {};
+    let initialCode = options.value;
+    ReactDOM.render(
+      <ToggleEditor
+        language={language}
+        initialCode={(initialCode == null) ? "" : initialCode}
+        api={api}
+        appElement={container}
+        options={options}
+        cmOptions={cmOptions}
+      />,
+      container
+    );
+    Object.assign(api, this.buildAPI());
+    // TODO(pcardune): What the heck is even going on here?
+    // constructors shouldn't return things that are not instances
+    // of the class they are constructing. We should not have to
+    // cast the return value to any for typescript to be happy.
+    return api as any;
+  }
+
+  buildAPI = () => {
+    return {
+      'fromTextArea': this.fromTextArea,
+    };
+  }
+
+  fromTextArea = myTextArea => {
+    myTextArea.parentNode.replaceChild(<ToggleEditor
+      appElement={myTextArea}
+      initialCode={myTextArea.value}
+    />, myTextArea);
+  }
+}
+export {
+  CodeMirrorBlocks,
+  Args,
+  DropTarget as DT,
```

Comment:
I understand using an acronym for "AST" given the context, but "DT" just seems like you're asking for trouble.

Reply:
Because of the "detox" reference, or because it's just a not a sufficiently large brevity win over `DropTarget`? I'm fine eliminating the short name.

Reply:
Just because "DT" is not very descriptive. At least when I see the letters DT, it makes me think of physics engines.

Reply:
You think physics, I think detox. I'm not sure what that says about me, but I'm happy to have you make this change. :)